lib/x11utils.pm

```diff
@@ -123,6 +123,7 @@ sub ensure_unlocked_desktop {
                 send_key 'esc';                                                                           # end screenlock
             };
         }
+        send_key 'shift';
```

Comment:
This should be handled as a product bug. Please, find https://progress.opensuse.org/issues/64812#Suggestions

It is a security issue that the screen lock doesn't hide the previous screen.
We should have a conditional record_soft_fail if this issue is detected.

Reply:
If you look at the ticket, the bug is already open.
I would add a conditional soft_failure if I knew how to reliably detect that bug

Reply:
In this branch: https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/10064/files#diff-f7cb81ed3ad3075100bf95d110f7e8adR75

It is detecting display-manager, in this branch it could check if the screen buffer is outdated by 
1. sending key
2. waiting still screen
3. then looping again

```perl
if (match_has_tag 'displaymanager') {
    send_key 'shift'; #<---------------------------------- here                                  
    wait_still_screen; #<--------------------------------- here
    next if (assert_screen('displaymanager', 0) != 0); #<- here

    if (check_var('DESKTOP', 'minimalx')) {
        type_string "$username";
        save_screenshot;
    }
    # Always select user if DM_NEEDS_USERNAME is set
    if ((!check_var('DESKTOP', 'gnome') || (is_sle('<15') || is_leap('<15.0'))) && !get_var('DM_NEEDS_USERNAME')) {
        send_key 'ret';
    }
    # On gnome, user may not be selected and using 'ret' is not enough in this case
    else {
        select_user_gnome($username);
    }
}
```

Reply:
With the screenbuffer frozen we do not know what will be asserted or did I get your point wrong?

Reply:
True, so maybe we need a handling before the loop, instead of handling for each if-branch.

Reply:
Which is basically what I am doing with the press of shift.
Of course we could add a wait_screen_change, but that would slow us down for the cases where everything works fine

Reply:
This what the shift is doing, but, unconditionally, and we need a condition to detect if it happens to record a soft fail

Reply:
Which leads as back to the problem, that it could be basically any screen
My best guess would be 'if not matched any other needle, assume softfail'

Reply:
go for it, I was just giving suggestions and requirements